### ISSUE TITLE
Multiple extension inside one class should not trigger CA1708

### ISSUE THREAD

**Issue by wertzui** [dotnet/roslyn#82494]
**Version Used**: 
.Net 10.0.2

**Steps to Reproduce**:
Two extension blocks within one class trigger the warning.
```csharp
public static class TestExtensions
{
    extension(int i)
    {
    }

    extension(string s)
    {
    }
}
```

> Names of 'Members' and '...TestExtensions.extension(int), ...TestExtensions.extension(string)' should differ by more than case

**Diagnostic Id**:
CA1708
Identifiers should differ by more than case

**Expected Behavior**:
No warning is triggered.

**Actual Behavior**:
A warning is triggered.

**Comment by MihuBot** [comment]
I'm a bot. Here are possible related and/or duplicate issues (I may be wrong):
- https://github.com/dotnet/sdk/issues/51716
- https://github.com/dotnet/roslyn/issues/78859
